liblib/CMakeLists.txt

```diff
@@ -4,194 +4,226 @@
-ENDIF()
-
-IF(ENABLE_S3_INTERNAL)
-  SET(liblib_LIBS ${liblib_LIBS} ncxml)
-ENDIF()
-
-IF(ENABLE_PLUGINS)
-  SET(liblib_LIBS ${liblib_LIBS} ncpoco)
-ENDIF()
-
-FOREACH(LIBS ${liblib_LIBS})
-  SET(LARGS ${LARGS} $<TARGET_OBJECTS:${LIBS}>)
-ENDFOREACH()
-
-ADD_LIBRARY(netcdf nc_initialize.c ${LARGS} )
-
-IF(STATUS_PARALLEL)
+target_sources(netcdf
+  PRIVATE
+    $<TARGET_OBJECTS:netcdf3>
+    $<TARGET_OBJECTS:dispatch>
+)
+
+if(USE_NETCDF4)
+  target_sources(netcdf
+    PRIVATE
+      $<TARGET_OBJECTS:netcdf4>
+  )
+endif()
+
+if(USE_HDF5)
+  target_sources(netcdf
+    PRIVATE
+      $<TARGET_OBJECTS:netcdfhdf5>
+  )
+endif()
+
+if(USE_PNETCDF)
+  target_sources(netcdf
+    PRIVATE
+      $<TARGET_OBJECTS:netcdfp>
+  )
+endif()
+
+if(USE_HDF4)
+  target_sources(netcdf
+    PRIVATE
+      $<TARGET_OBJECTS:netcdfhdf4>
+  )
+endif()
+
+if(ENABLE_DAP2)
+  target_sources(netcdf
+    PRIVATE
+      $<TARGET_OBJECTS:oc2>
+      $<TARGET_OBJECTS:dap2>
+  )
+endif()
+
+if(ENABLE_DAP4)
+  target_sources(netcdf
+    PRIVATE
+      $<TARGET_OBJECTS:dap4>
+      $<TARGET_OBJECTS:ncxml>
+  )
+endif()
+
+if(ENABLE_NCZARR)
+  target_sources(netcdf
+    PRIVATE
+      $<TARGET_OBJECTS:nczarr>
+  )
+endif()
+
+if(ENABLE_S3_INTERNAL)
+  target_sources(netcdf
+    PRIVATE
+      $<TARGET_OBJECTS:ncxml>
+  )
+endif()
+
+if(ENABLE_PLUGINS)
+  target_sources(netcdf
+    PRIVATE
+      $<TARGET_OBJECTS:ncpoco>
+  )
+endif()
+
+target_sources(netcdf
+  PRIVATE
+    ${CMAKE_CURRENT_SOURCE_DIR}/nc_initialize.c 
+)
+
+if(STATUS_PARALLEL)
   # TODO: Make PUBLIC when other dependencies have PUBLIC/PRIVATE specified
   target_link_libraries(netcdf MPI::MPI_C)
-ENDIF(STATUS_PARALLEL)
+endif(STATUS_PARALLEL)
 
-IF(TARGET MPI::MPI_C)
+if(TARGET MPI::MPI_C)
   target_link_libraries(netcdf MPI::MPI_C)
-ENDIF(TARGET MPI::MPI_C)
+endif(TARGET MPI::MPI_C)
 
-IF(MOD_NETCDF_NAME)
-  SET_TARGET_PROPERTIES(netcdf PROPERTIES LIBRARY_OUTPUT_NAME ${NETCDF_LIB_NAME})
-  SET_TARGET_PROPERTIES(netcdf PROPERTIES ARCHIVE_OUTPUT_NAME ${NETCDF_LIB_NAME})
-  SET_TARGET_PROPERTIES(netcdf PROPERTIES RUNTIME_OUTPUT_NAME ${NETCDF_LIB_NAME})
-ENDIF()
+if(MOD_NETCDF_NAME)
+  set_target_properties(netcdf PROPERTIES LIBRARY_OUTPUT_NAME ${NETCDF_LIB_NAME})
+  set_target_properties(netcdf PROPERTIES ARCHIVE_OUTPUT_NAME ${NETCDF_LIB_NAME})
+  set_target_properties(netcdf PROPERTIES RUNTIME_OUTPUT_NAME ${NETCDF_LIB_NAME})
+endif()
 
 # Make sure that netcdf.dll path does not include the build configuration
-IF(MSVC)
-  SET_TARGET_PROPERTIES(netcdf PROPERTIES
-	RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-	RUNTIME_OUTPUT_DIRECTORY_DEBUG ${CMAKE_CURRENT_BINARY_DIR}
-	RUNTIME_OUTPUT_DIRECTORY_RELEASE ${CMAKE_CURRENT_BINARY_DIR})
-ENDIF()
+if(MSVC)
+  set_target_properties(netcdf PROPERTIES
+    RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+    RUNTIME_OUTPUT_DIRECTORY_DEBUG ${CMAKE_CURRENT_BINARY_DIR}
+    RUNTIME_OUTPUT_DIRECTORY_RELEASE ${CMAKE_CURRENT_BINARY_DIR})
```

Comment:
This should not be needed

Reply:
I think it is, otherwise msvc will place the output in separate Debug or Release directories